test/audits/link-with-unclear-purpose.js

```diff
@@ -27,6 +27,67 @@ test('a link with meaningful text is good', function() {
         { elements: [], result: axs.constants.AuditResult.PASS });
 });
 
+test('a link with an img with meaningful alt text is good', function() {
+    var fixture = document.getElementById('qunit-fixture');
+    var a = document.createElement('a');
+    a.href = '#main';
+    var img = a.appendChild(document.createElement('img'));
+    img.setAttribute('alt', 'Learn more about trout fishing');
+
+    fixture.appendChild(a);
+    var rule = axs.AuditRules.getRule('linkWithUnclearPurpose');
+    deepEqual(
+        rule.run({scope: fixture}),
+        { elements: [], result: axs.constants.AuditResult.PASS });
+});
+
+test('a link with an img with meaningless alt text is bad', function() {
+    var fixture = document.getElementById('qunit-fixture');
+    var a = document.createElement('a');
+    a.href = '#main';
+    var img = a.appendChild(document.createElement('img'));
+    img.setAttribute('alt', 'Click here!');
+
+    fixture.appendChild(a);
+    var rule = axs.AuditRules.getRule('linkWithUnclearPurpose');
+    deepEqual(rule.run({scope: fixture}),
+        { elements: [a], result: axs.constants.AuditResult.FAIL });
+});
+
+test('a link with meaningful aria-label is good', function() {
+    var fixture = document.getElementById('qunit-fixture');
+    // Style our link to be visually meaningful with no descendent nodes at all.
+    fixture.innerHTML = '<style>a.trout::after{content: "Learn more about trout fishing"}</style>';
```

Comment:
Ok, it looks like they do. I'll file a bug - don't worry about it in this change, but it might be worth making a note here that this test will need to be updated.